internal/datasource/config/config_writer_test.go

```diff
@@ -17,23 +19,20 @@ import (
 
 	"github.com/nginx/agent/v3/api/grpc/instances"
 	"github.com/nginx/agent/v3/internal/client"
-	datasource_os "github.com/nginx/agent/v3/internal/datasource/os"
+	"github.com/nginx/agent/v3/internal/datasource/nginx"
 
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func TestUpdateInstanceConfig(t *testing.T) {
+func TestWriteConfig(t *testing.T) {
```

Comment:
Just checked out to your branch and ran the unit tests, looks like the `Write()` function is not being tested for cases where latest version is not already on disk. Think we should make sure we cover those too.

Also, the code coverage percentage (see bottom of screenshot) is skewed, because the coverage calculation now includes our mock code, which we obviously aren't going to unit test.

![image](https://github.com/nginx/agent/assets/85518764/a5459d75-cad7-430e-861a-6d70d7c68e5f)